pcf/particle/aws/glacier/glacier_vault.py

```diff
@@ -0,0 +1,140 @@
+from pcf.core import State
+from pcf.util import pcf_util
+
+logger = logging.getLogger(__name__)
+
+class GlacierVault(AWSResource):
+    """
+    This is the implementation of Amazon's Glacier service. Additional functions that are callable on this particle
+    are upload_archive, initiate_job, list_jobs, delete_archive, delete_job, add_tags_to_vault, list_tags_for_vault
+    """
+    flavor = "glacier"
+
+    state_lookup = {
+        "missing": State.terminated,
+        "active": State.running,
+        "inactive": State.terminated
+    }
+
+    equivalent_states = {
+        State.running: 1,
+        State.stopped: 0,
+        State.terminated: 0
+    }
+
+    def __init__(self, particle_definition):
+        super(GlacierVault, self).__init__(
+            particle_definition=particle_definition,
+            resource_name="glacier",
+        )
+        self.vault_name = self.desired_state_definition.get("vaultName")
+        self.account_id = self.desired_state_definition.get("accountId", "-")
+
+    def _terminate(self):
+        """
+        Deletes the Glacier vault if vault is empty
+
+        Returns:
+             response of boto3 delete_vault
+        """
+        return self.client.delete_vault(vaultName=self.vault_name)
+
+    def _start(self):
+        """
+        Creates the Glacier vault
+
+        Returns:
+             response of boto3 create_vault
+        """
+        response = self.client.create_vault(
+            vaultName=self.vault_name,
+            accountId=self.account_id
+        )
+
+        if self.custom_config.get("Tags"):
+            tags = self.custom_config.get("Tags")
+
+            self.client.add_tags_to_vault(
+                vaultName=self.vault_name,
+                Tags=tags
+            )
+        return response
+
+
+    def _stop(self):
+        """
+        Glacier does not have a stopped state so it calls terminate.
+        """
+        return self.terminate()
+
+    def get_status(self):
+        """
+        Determines if the vault exists
+
+        Returns:
+             status (dict)
+        """
+
+        try:
+            vault_object = self.client.describe_vault(
+                vaultName=self.vault_name,
+                accountId=self.account_id
+            )
+        except ClientError as e:
+            logger.info("{}. State is missing".format(e))
+            return {"status":"missing"}
+        return {"status":"active"}
+
+    def sync_state(self):
+        """
+        Uses get_status() to determine whether the vault exists or not and sets the current state definition
+
+        Returns:
+            void
+        """
+
+        full_status = self.get_status()
+
+        if full_status:
+            status = full_status.get("status", "missing").lower()
+            self.state = self.state_lookup.get(status)
+
+            # set current state definition
+            self.current_state_definition = self.desired_state_definition
```

Comment:
this should equal status.

Reply:
this would just be changed to 

```
if not full_status:
            self.state = self.state_lookup.get("missing")
else:
           self.state=self.state_lookup.get("active")
self.current_state_definition = full_status


```